### PR TITLE
Update default etcd server to 3.2 for kubernetes 1.11

### PR DESCRIPTION
**Pull request by jpbetz**
Repply #59836 but with latest etcd 3.2 patch version (3.2.18 which includes mvcc fix and leader election timeout fix) and default `--snapshot-count` to 10k to resolve performance regression in previous etcd 3.2 server upgrade attempt (https://github.com/kubernetes/kubernetes/issues/60589#issuecomment-372458578).

See https://github.com/kubernetes/kubernetes/issues/60589#issuecomment-372458578 for details on the root cause of the performance regression and scalability test results of setting `--snapshot-count` to 10k.

```release-note
Upgrade the default etcd server version to 3.2.18
```
@gyuho @shyamjvs @jdumars @timothysc

### CODE REVIEW

**Comment by jpbetz** [comment]
@jdumars, @shyamjvs Is there time for this to make this in to k8s 1.10?

**Comment by timothysc** [comment]
Has this been scale tested?  Also what are the details on the fix and it's root cause?

File: cluster/gce/manifests/etcd.manifest [hunk 1]
```diff
@@ -39,6 +39,9 @@
       },
       { "name": "ETCD_CREDS",
         "value": "{{ etcd_creds }}"
+      },
+      { "name": "ETCD_SNAPSHOT_COUNT",
+        "value": "10000"
```
Comment (timothysc) [inline cluster/gce/manifests/etcd.manifest]: You didn't set this default on the kubeadm change.

**Comment by shyamjvs** [comment]
I would say it would be safer to get this in for next release, given:
- it's a relatively high risk change to be made this close to release (we don't yet know if there's sth other than snapshot-count)
- we may lack bandwidth to scale test this change with the debugging happening in #60589 

How critical is this (given that some key fixes have already been backported to 3.1)?

**Comment by jpbetz** [comment]
@timothysc Yes, see https://github.com/kubernetes/kubernetes/issues/60589#issuecomment-372458578 for details on the root cause of the performance regression and scalability test results of setting `--snapshot-count` to 10k.

File: cluster/gce/manifests/etcd.manifest (hunk 1 above)
Comment (jpbetz) [reply cluster/gce/manifests/etcd.manifest]: Added to `cmd/kubeadm/app/phases/etcd/local.go`

**Comment by jpbetz** [comment]
@shyamjvs Setting milestone to v1.11

**Comment by jpbetz** [comment]
Published k8s.gcr.io images for etcd and etcd-empty-dir-cleanup

**Review by timothysc — CHANGES_REQUESTED** [review]
Please squash these patches then lgtm from me.  

Let's get this done early in the cycle to prevent another last minute issue. 

/cc @shyamjvs @wojtek-t @jdumars

**Review by timothysc — APPROVED** [review]
/lgtm 
/approve

**Comment by wojtek-t** [comment]
There is a hypothesis that his hurts scalability (as I mentioned in #62808 couple days ago).
For now, we will temporarily revert our scalability jobs to use 3.1.12 etcd to verify that.
If that's the case, we will need to think what to do about this PR.

@jpbetz @timothysc @lavalamp  - FYI

**Comment by jpbetz** [comment]
Thanks for the heads up @wojtek-t

**Comment by liggitt** [comment]
@gyuho fyi

**Comment by gyuho** [comment]
ack

**Pull request merged**